sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/exception/CosmosExceptionUtils.java

```diff
@@ -37,9 +37,33 @@ public static <T> Mono<T> exceptionHandler(String message, Throwable throwable,
         }
         //  Unwrap the exception in case if it is a reactive exception
         final Throwable unwrappedThrowable = Exceptions.unwrap(throwable);
-        if (unwrappedThrowable instanceof CosmosException) {
-            CosmosException cosmosException = (CosmosException) unwrappedThrowable;
+        if (unwrappedThrowable instanceof CosmosException cosmosException) {
             CosmosUtils.fillAndProcessCosmosExceptionDiagnostics(responseDiagnosticsProcessor, cosmosException);
+
+            switch (unwrappedThrowable.getClass().getSimpleName()) {
```

Comment:
Instead of relying on the name of the exception, we should rely on the status code. 
Although, the status codes are also implementation details, but at least that is a stronger contract compared to the name of the exception class.

Reply:
@kushagraThapar we cannot rely on the status codes, for example BadRequestException and NonStreamingOrderByBadRequestException both throw the same status code so that would not allow us to differentiate.
![image](https://github.com/user-attachments/assets/24538216-6977-4841-92bd-21d9f981c3af)
![image](https://github.com/user-attachments/assets/192d89ef-3696-4600-a567-8e3318653a09)

Reply:
And using the name is super brittle - this definitely seems wrong.

Reply:
If you want multiple exceptions per status code they should be distinguishable by statusCode + substatuscode - any differences in exception with same tuple of statuscode7substatuscode can be merged (and honestly are a bad idea even in core SDK)

Reply:
@FabianMeiswinkel so then why does azure-cosmos have multiple exceptions for the same status code? And exceptions can also have multiple sub-status codes so there is not a 1-1 mapping there either. If there is some mapping somewhere of codes I can use that but so far I do not see any consistent mapping